templates/nginx-static.j2

```diff
@@ -13,33 +16,39 @@ upstream auth_backend {
 
 server {
     server_name {{ domain }};
-
     listen {{ port }};
     listen [::]:{{ port }};
 
     root {{ document_root }};
 
-    location / {
+    location /ubuntu/pool/ {
+        autoindex on;
       {%- if auth_backends %}
         auth_request /auth;
       {%- else %}
         auth_basic "{{ domain }} archive";
         auth_basic_user_file {{ basic_auth_file }};
       {%- endif %}
+    }
+
+    location / {
         autoindex on;
     }
 
   {% if auth_backends %}
     location = /auth {
         internal;
-        proxy_pass http://auth_backend/auth-check/;
-        proxy_set_header Original-URI $request_uri;
-      {%- if auth_cache_time %}
+      {%- if auth_cache_enabled %}
+        # Activate cache, for both ok and unauthorized auth responses.
+        # This way we avoid contacting the contracts service for every
+        # package when executing "apt upgrade". The cache key is the
+        # HTTP basic auth header.
         proxy_cache auth;
+        proxy_cache_valid 200 401 {{ auth_cache_duration }};
```

Comment:
Just checking out loud, it's definitely wise to cache 401s.
There is a case where:
1. User tries to use ESM. Unauthorised.
2. Buys ESM and tries again

The token will have changed therefore won't be in cache, yes?

Reply:
That seems correct to me.